src/main/java/dev/slimevr/filtering/QuaternionMovingAverage.java

```diff
@@ -0,0 +1,84 @@
+package dev.slimevr.filtering;
+
+import com.jme3.math.FastMath;
+import com.jme3.math.Quaternion;
+
+
+public class QuaternionMovingAverage {
+
+	private final float smoothFactor;
+	private final float predictFactor;
+	private final CircularArrayList<Quaternion> rotBuffer;
+	private final Quaternion quatBuf = new Quaternion();
+	private final Quaternion targetQuat = new Quaternion();
+	private final Quaternion lastQuaternion;
+	private final Quaternion filteredQuaternion;
+	private static final float SMOOTH_QUADRATIC_DIVIDER = 11f;
+	private static final float SMOOTH_QUADRATIC_MIN = 0.022f;
+	private static final float PREDICT_QUADRATIC_DIVIDER = 3.1f;
+	private static final float PREDICT_QUADRATIC_MIN = 0.11f;
```

Comment:
The heck are these? Maybe it could have a comment explaining it, or at least some sort of link to what it is?